### ISSUE TITLE
[UI] visualization of failed pipelines inaccurate after upgrading argo

### ISSUE THREAD

**Issue by Bobgy** [kubeflow/pipelines#5322]
### What steps did you take:
[A clear and concise description of what the bug is.]

### What happened:
It seems there was a change that all skipped nodes in the pipeline are also added to workflow status, so they also render in KFP UI now. However, it doesn't show properly.

TODO: I'll upload a screenshot and give an example here.

### What did you expect to happen:
The skipped nodes should either be hidden, or we remove the unfinished sign after the last unfinished node.

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?

KFP version: 

KFP SDK version: 

### Anything else you would like to add:
[Miscellaneous information that will assist in solving the issue.]

/kind bug

**Comment by Bobgy** [comment]
Example pipeline

```
HOST = '<redacted>'

import kfp
import kfp.components as comp


def fail():
    import sys
    println("failed")
    sys.exit(1)


def echo():
    println("hello world")


def main():
    fail_op = comp.func_to_container_op(fail)
    echo_op = comp.func_to_container_op(echo)

    @kfp.dsl.pipeline(name='fail sample')
    def fail_sample():
        fail_task = fail_op()
        echo_task = echo_op()
        echo_task.after(fail_task)

    client = kfp.Client(host=HOST)
    client.create_run_from_pipeline_func(fail_sample, {})


if __name__ == "__main__":
    main()
```

**Comment by Bobgy** [comment]
However, here is the pipeline visualization of the failed pipeline

![image](https://user-images.githubusercontent.com/4957653/111728523-fec26200-88a7-11eb-8ecd-d8c4508af8f5.png)

Expected: there should be no unfinished sign after the last step

**Comment by Bobgy** [comment]
Fixed by #5339 
@zijianjoy FYI, if you use `Fixes #5322` in your PR's title. This issue will be auto closed when your PR is merged. That's best practice for keeping the issue lifecycle up-to-date.
